src/Generators/Microsoft.Gen.AutoClient/DiagDescriptors.cs

```diff
@@ -11,136 +11,134 @@ internal sealed class DiagDescriptors : DiagDescriptorsBase
-        id: "R9G310",
+        id: "AUTOCLIENTGEN008",
         title: Resources.ErrorMissingMethodAttributeTitle,
         messageFormat: Resources.ErrorMissingMethodAttributeMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorInterfaceIsGeneric { get; } = Make(
-        id: "R9G311",
+        id: "AUTOCLIENTGEN009",
         title: Resources.ErrorInterfaceIsGenericTitle,
         messageFormat: Resources.ErrorInterfaceIsGenericMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorInterfaceName { get; } = Make(
-        id: "R9G312",
+        id: "AUTOCLIENTGEN010",
         title: Resources.ErrorInterfaceNameTitle,
         messageFormat: Resources.ErrorInterfaceNameMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorDuplicateBody { get; } = Make(
-        id: "R9G313",
+        id: "AUTOCLIENTGEN011",
         title: Resources.ErrorDuplicateBodyTitle,
         messageFormat: Resources.ErrorDuplicateBodyMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorMissingParameterUrl { get; } = Make(
-        id: "R9G314",
+        id: "AUTOCLIENTGEN012",
         title: Resources.ErrorMissingParameterUrlTitle,
         messageFormat: Resources.ErrorMissingParameterUrlMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorDuplicateCancellationToken { get; } = Make(
-        id: "R9G315",
+        id: "AUTOCLIENTGEN013",
         title: Resources.ErrorDuplicateCancellationTokenTitle,
         messageFormat: Resources.ErrorDuplicateCancellationTokenMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorMissingCancellationToken { get; } = Make(
-        id: "R9G315",
+        id: "AUTOCLIENTGEN014",
         title: Resources.ErrorMissingCancellationTokenTitle,
         messageFormat: Resources.ErrorMissingCancellationTokenMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorPathWithQuery { get; } = Make(
-        id: "R9G316",
+        id: "AUTOCLIENTGEN015",
         title: Resources.ErrorPathWithQueryTitle,
         messageFormat: Resources.ErrorPathWithQueryMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorDuplicateRequestName { get; } = Make(
-        id: "R9G317",
+        id: "AUTOCLIENTGEN016",
         title: Resources.ErrorDuplicateRequestNameTitle,
         messageFormat: Resources.ErrorDuplicateRequestNameMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorInvalidHttpClientName { get; } = Make(
-        id: "R9G318",
+        id: "AUTOCLIENTGEN017",
         title: Resources.ErrorInvalidHttpClientNameTitle,
         messageFormat: Resources.ErrorInvalidHttpClientNameMessage,
         category: Category);
 
     public static DiagnosticDescriptor ErrorInvalidDependencyName { get; } = Make(
-        id: "R9G319",
+        id: "AUTOCLIENTGEN018",
```

Comment:
NIT: There is no real convention I believe, but it would be good to try to simplify these Ids if possible.